app/src/main/java/org/openedx/app/AppViewModel.kt

```diff
@@ -120,4 +123,11 @@ class AppViewModel(
             }
         }
     }
+
+    fun markNotificationAsRead(notificationId: Int) {
+        viewModelScope.launch {
+            val marked = interactor.markNotificationAsRead(notificationId)
+            Log.d("AppViewModel", "Notification marked as read: $marked")
```

Comment:
We need a try-catch block for this and send a non-fatal error in the catch block.

Reply:
addressed.